snowshu/core/graph.py

```diff
@@ -91,6 +93,46 @@ def build_graph(self, configs: Configuration) -> None:
             f'Identified a total of {len(self.graph)} relations to sample based on the specified configurations.')
 
         if not networkx.algorithms.is_directed_acyclic_graph(self.graph):
+            graph_cycles = list(networkx.simple_cycles(self.graph))
```

Comment:
This method have become really huge - maybe put some (or even all) logic related to presence of graph_cycles into separate method?